packages/toolbar/src/browser/toolbar.tsx

```diff
@@ -234,23 +232,11 @@ export class ToolbarImpl extends TabBarToolbar {
         );
     }
 
-    protected renderItemWithDraggableWrapper(item: ToolbarItem, position: ToolbarItemPosition): React.ReactNode {
+    protected renderItemWithDraggableWrapper(item: TabBarToolbarItem, position: ToolbarItemPosition): React.ReactNode {
         const stringifiedPosition = JSON.stringify(position);
         let toolbarItemClassNames = '';
-        let renderBody: React.ReactNode;
+        const renderBody = item.render(this);
 
-        if (!ReactTabBarToolbarItem.is(item)) {
-            toolbarItemClassNames = TabBarToolbar.Styles.TAB_BAR_TOOLBAR_ITEM;
-            if (this.evaluateWhenClause(item.when)) {
-                toolbarItemClassNames += ' enabled';
```

Comment:
Without adding 'enabled',  toolbar items are rendered as disabled

Reply:
That is the responsibility of the toolbar item.

Reply:
Then we are missing some implementation, cause the test is failing because `workbench.action.showCommands` toolbar item is not enabled.